### ISSUE TITLE
fortpy compiler not recognizing allocatable tag in test for find_permutation_of_group_and_dset

### ISSUE THREAD

**Issue by wsmorgan** [msg-byu/enumlib#3]
I've got the xml here:
  <decorates name="enumeration_utilities.find_permutation_of_group_and_dset">
    <group name="enum_utils" purpose="testing_fortpy_bug_allocatable" staging="../unittests/">
      <test identifier="find_permutation_of_group_and_dset"
        description="Original tests of mapping structure to real space." cases="[1-129]"
        timed="true">
    <output folder="../tests/enumeration_utilities"
        file="find_permutation_of_group_and_dset_perm.out.{}"
        template="integer.xml" identifier="perm.out" />
    <target name="perm" compareto="perm.out" varfile="perm.out" when="each"/>
      </test>
      <assignment name="g" value="gfile" allocate="true">
    <value identifier="gfile" file="find_permutation_of_group_and_dset_g.in.{}"
           folder="../tests/enumeration_utilities" rename="g.in"/>
      </assignment>
      <assignment name="gp" value="gpfile" allocate="true">
    <value identifier="gpfile" file="find_permutation_of_group_and_dset_gp.in.{}"
           folder="../tests/enumeration_utilities" rename="gp.in"/>
      </assignment>
      <assignment name="d" value="dfile" allocate="true">
    <value identifier="dfile" file="find_permutation_of_group_and_dset_d.in.{}"
           folder="../tests/enumeration_utilities" rename="d.in"/>
      </assignment>
      <assignment name="perm" value="permfile">
    <value identifier="permfile" file="find_permutation_of_group_and_dset_perm.in.{}"
           folder="../tests/enumeration_utilities" rename="perm.in"/>
      </assignment>
    </group>
  </decorates>

The compiler fails because the parser isn't adding the allocatable tag to the variables.

**Comment by rosenbrockc** [comment]
Could you post the XML again and put it in a code block? The XML got clobbered.

**Comment by wsmorgan** [comment]
Right sorry. Here it is again.

```
<decorates name="enumeration_utilities.find_permutation_of_group_and_dset">
    <group name="enum_utils" purpose="testing_fortpy_bug_allocatable" staging="../unittests/">
      <test identifier="find_permutation_of_group_and_dset"
        description="Original tests of mapping structure to real space." cases="[1-129]"
        timed="true">
    <output folder="../tests/enumeration_utilities"
        file="find_permutation_of_group_and_dset_perm.out.{}"
        template="integer.xml" identifier="perm.out" />
    <target name="perm" compareto="perm.out" varfile="perm.out" when="each"/>
      </test>
      <assignment name="g" value="gfile" allocate="true">
    <value identifier="gfile" file="find_permutation_of_group_and_dset_g.in.{}"
           folder="../tests/enumeration_utilities" rename="g.in"/>
      </assignment>
      <assignment name="gp" value="gpfile" allocate="true">
    <value identifier="gpfile" file="find_permutation_of_group_and_dset_gp.in.{}"
           folder="../tests/enumeration_utilities" rename="gp.in"/>
      </assignment>
      <assignment name="d" value="dfile" allocate="true">
    <value identifier="dfile" file="find_permutation_of_group_and_dset_d.in.{}"
           folder="../tests/enumeration_utilities" rename="d.in"/>
      </assignment>
      <assignment name="perm" value="permfile">
    <value identifier="permfile" file="find_permutation_of_group_and_dset_perm.in.{}"
           folder="../tests/enumeration_utilities" rename="perm.in"/>
      </assignment>
    </group>
  </decorates>
```

**Comment by rosenbrockc** [comment]
This bug is related to a dimension parsing bug in fortpy. It is fixed starting in Revision 1.6.0.
